scripts/parser/backend/cmakecachestring.py

```diff
@@ -0,0 +1,41 @@
+import re
+
+class cmakecache_string(str):
+   """
+   simple utility string that abstracts away some details from the parsing
+   process
+   """
+   regex_expression =  r"^([A-Za-z_0-9]+?):([A-Za-z]+?)=(.*)$"
+
+   def user_comment(self) -> str:
+      """
+      if a line starts with // we assume it's a comment, and we strip away the
+      newline that is added by CMake
+      """
+      return self[2:].rstrip()
+
+   def is_comment(self) -> bool:
+      return self[:2] == "//"
+
+   def is_empty(self) -> bool:
+      """
+      lenght assumption is safe in this case because of the CMakeCache format
+      A comment has at least lenght 2, while a variable has always a name and
+      an explicit type
+      """
+      return len(self) <= 1
+
+   def split_variable(self):
```

Comment:
**Misleading name**: This should really be something like `parse_variable`.

Reply:
changed.